tests/components/azure_data_explorer/test_config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Test the AEH config flow."""
+from azure.kusto.data.exceptions import KustoAuthenticationError, KustoServiceError
+import pytest
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.azure_data_explorer.const import DOMAIN
+
+from .const import BASE_CONFIG, UPDATE_OPTIONS
+
+
+async def test_config_flow(hass, mock_setup_entry) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}, data=None
+    )
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["errors"] == {}
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        BASE_CONFIG.copy(),
+    )
+
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == "cluster.region.kusto.windows.net"
+    mock_setup_entry.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    ("test_input", "expected"),
+    [
+        (KustoServiceError("test"), "cannot_connect"),
+        (KustoAuthenticationError("test", Exception), "invalid_auth"),
+        (Exception(), "unknown"),
+    ],
+)
+async def test_config_flow_errors(
+    test_input,
+    expected,
+    hass,
+    mock_execute_query,
+) -> None:
+    """Test we handle connection KustoServiceError."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+        data=None,
+    )
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["errors"] == {}
+
+    mock_execute_query.side_effect = test_input
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        BASE_CONFIG.copy(),
+    )
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": expected}
```

Comment:
Config flow tests should end with either creating an entry or aborting the flow.
Please extend the test that it is possible to recover form the error

Reply:
This suggestion was not implemented.
On the end of each config flow test should end with `FlowResultType.ABORT` or `FlowResultType.CREATE_ENTRY`.
Currently you test if the form is open at the end.

Please adjust all tests

Reply:
If the configflow experiences an error here, it just reopens the form, so I do not get an abort or create

Reply:
That's correct but you should test that the user can fix the error and create an entry as example